Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -742,6 +741,10 @@ public class BrowserViewController: UIViewController {
     }
   }
   
+  @objc func appWillTerminateNotification() {
+    tabManager.saveAllTabs()
```

Comment:
Is there a possibility here app can terminate itself before `saveAllTabs` executed fully ?

Reply:
Not really because it is saved on the main-thread. Also, this is only to save scroll-position. Tabs are already automatically saved the moment a page is loaded and the moment the tab is created.